props/blaster_BC_buttons.h

```diff
@@ -0,0 +1,720 @@
+  }
+}
+
+void Loop() override {
+  PropBase::Loop();
+  PollNextAction();
+  sound_library_.Poll(wav_player);
+  if (wav_player && !wav_player->isPlaying()) wav_player.Free();
+
+  if (blaster_mode == MODE_AUTO) {
+    // First time through, let bgnauto finish first if we have one
+    if (SFX_bgnauto && GetWavPlayerPlaying(&SFX_bgnauto)) return;
+
+    // No bgnauto or bgnauto has finished playing, so initialize and start auto-firing
+    if (auto_firing_ && !auto_player_) {
+      auto_player_ = GetWavPlayerPlaying(&SFX_auto);
+      shots_fired_++;
+      auto_time_ = millis();
+      PVLOG_NORMAL << "******** AUTOFIRING - Remaining shots = " << GetBulletCount() << "\n";
+    }
+    // use auto.wav's length to count shots
+    if (auto_firing_ && auto_player_ && (millis() - auto_time_) > auto_sound_length) {
+      auto_sound_length = auto_player_->length() * 1000;
+      shots_fired_++;
+      if (IsEmpty()) {
+        SaberBase::DoEndLockup();
+        SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+        auto_firing_ = false;
+        auto_player_.Free();
+        auto_time_ = millis(); // Reset for the first empty sound
+        return;
+      }
+      auto_time_ = millis();
+      PVLOG_NORMAL << "******** AUTOFIRING - Remaining shots = " << GetBulletCount() << "\n";
+    }
+
+    // This will mimic the hammer running on no bullets at the same rate it was firing.
+    // if there are different length auto.wavs, the last played's length is used.
+    if (!auto_firing_ && IsEmpty() && trigger_is_pressed_ && (millis() - auto_time_) > auto_sound_length) {
+      SaberBase::DoEffect(EFFECT_EMPTY, 0);
+      auto_time_ = millis(); // Reset for the next empty sound
+    }
+  }
+}
+
+  void Reload() override {
+    SaberBase::DoEffect(EFFECT_RELOAD, 0);
+    shots_fired_ = 0;
+    is_jammed_ = false;
+  }
+
+  void ClipOut() override {
+    if (max_shots_ != -1) shots_fired_ = max_shots_;
+    SaberBase::DoEffect(EFFECT_CLIP_OUT, 0);
+    no_clip_ = true;
+  }
+
+  void ClipIn() override {
+    SaberBase::DoEffect(EFFECT_CLIP_IN, 0);
+    shots_fired_ = 0;
+    is_jammed_ = false;
+    no_clip_ = false;
+  }
+
+  // Pull in parent's SetPreset, but turn the blaster on.
+  void SetPreset(int preset_num, bool announce) override {
+    PropBase::SetPreset(preset_num, announce);
+      // Just always default to on.
+       if (!SaberBase::IsOn()) On();
+  }
+
+  void PollNextAction() override {
+    if (millis() - time_base_ > next_event_time_) {
+      switch (next_action_) {
+        case NEXT_ACTION_NOTHING:
+          break;
+        case NEXT_ACTION_BLOW:
+          Off(OFF_BLAST);
+          break;
+      }
+      next_action_ = NEXT_ACTION_NOTHING;
+    }
+  }
+
+
+  void selfDestruct() override {
+        SaberBase::DoEffect(EFFECT_DESTRUCT, 0);
+    float len = hybrid_font.GetCurrentEffectLength();
+    SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+      SetNextActionF(NEXT_ACTION_BLOW, len);
+  }
+// Volume Menu
+    void VolumeMenu() {
```

Comment:
Use a mode?

Reply:
Yes, now I need to update this ancient PR.

Reply:
done